lib/avalon/transcript_parser.rb

```diff
@@ -0,0 +1,70 @@
+# Copyright 2011-2024, The Trustees of Indiana University and Northwestern
+#   University.  Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+# 
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software distributed
+#   under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+#   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#   specific language governing permissions and limitations under the License.
+# ---  END LICENSE_HEADER BLOCK  ---
+
+require 'avalon/docx_file'
+
+module Avalon
+  class TranscriptParser
+    TEXT_TYPE = ['text/vtt', 'text/srt', 'text/plain', 'application/vnd.openxmlformats-officedocument.wordprocessingml.document']
+
+    # Passed in transcript_file must be instance of ActiveStorage::Attached::One
+    def initialize(transcript_file)
+      @mime_type = transcript_file.content_type
+      @transcript = transcript_file.download
+    end
+
+    def transcript_plaintext
```

Comment:
Now that that `TranscriptParser` is a class which wraps a single transcript what do you think about naming this method `plaintext` and using it to memoize an instance variable?  Something like:
```
def plaintext
  unless @plaintext
    @plaintext = Avalon::DocxFile.new(@transcript).unformatted_text if @mime_type == 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'
    @plaintext ||= @transcript
  end
  @plaintext
end
```